### PR TITLE
feat: add o1 model

### PR DESCRIPTION
**Pull request by skymkmk**
#### 💻 变更类型 | Change Type

- [X] feat    
- [ ] fix    
- [ ] refactor    
- [ ] perf    
- [ ] style    
- [ ] test    
- [ ] docs    
- [ ] ci    
- [ ] chore    
- [ ] build    

#### 🔀 变更说明 | Description of Change

添加了 o1-mini 与 o1-preview 模型

#### 📝 补充信息 | Additional Information

o1 系列模型暂时不支持多模态，system 角色，流式传输，工具函数（plugin 插件），温度，核采样，两个惩罚度等，OpenAI 可能会在下面几周添加这些功能的支持。

## Summary by CodeRabbit

- **New Features**
	- Introduced support for new models "o1-mini" and "o1-preview" with updated knowledge cut-off dates.
	- Enhanced message processing logic for the "o1" model, including exclusion of system messages and fixed parameters for improved performance.
	- Adjusted request timeout settings to optimize processing time for the "o1" model.

- **Bug Fixes**
	- Improved handling of model-specific configurations to ensure accurate message processing.

### CODE REVIEW

File: app/client/platforms/openai.ts [hunk 1]
```diff
@@ -181,17 +182,19 @@ export class ChatGPTApi implements LLMApi {
         const content = visionModel
           ? await preProcessImageContent(v.content)
           : getMessageTextContent(v);
-        messages.push({ role: v.role, content });
+        if(!(isO1 && v.role === "system"))
+          messages.push({ role: v.role, content });
       }
 
+      // O1 not support image, tools (plugin in ChatGPTNextWeb) and system, stream, logprobs, temperature, top_p, n, presence_penalty, frequency_penalty yet.
```
Comment (lloydzhou) [inline app/client/platforms/openai.ts:189]: `shouldStream` will depend on `isO1` (see line: 210)

**Review by lloydzhou — CHANGES_REQUESTED** [review]

File: app/client/platforms/openai.ts (hunk 1 above)
Comment (skymkmk) [reply app/client/platforms/openai.ts:189]: Sorry, it seems there were some merge issues when migrating the code from my private repository. I will add it later.

Comment (skymkmk) [reply app/client/platforms/openai.ts:189]: fix it in d0dce654bffead1971600a5feb313d9079800254

**Review by lloydzhou — APPROVED** [review]

**Comment by skymkmk** [comment]
Is the default timeout a bit too short? When using the o1 model, my average time is generally over 60 seconds, with an average of around 90 seconds. Should the timeout threshold be increased for o1?

File: app/client/platforms/openai.ts [hunk 2]
```diff
@@ -181,17 +182,19 @@ export class ChatGPTApi implements LLMApi {
         const content = visionModel
           ? await preProcessImageContent(v.content)
           : getMessageTextContent(v);
-        messages.push({ role: v.role, content });
+        if(!(isO1 && v.role === "system"))
+          messages.push({ role: v.role, content });
```
Comment (Dogtiti) [inline app/client/platforms/openai.ts:186]: 这里感觉处理的不太合理，为什么过滤system message

**Review by Dogtiti — CHANGES_REQUESTED** [review]

File: app/client/platforms/openai.ts (hunk 2 above)
Comment (skymkmk) [reply app/client/platforms/openai.ts:186]: 因为 o1 目前不支持 system role

Comment (Dogtiti) [reply app/client/platforms/openai.ts:186]: 如果转换为 user message:   role=user 呢？

Comment (skymkmk) [reply app/client/platforms/openai.ts:186]: 个人感觉没必要

1. O1 很贵，每次都附加 system 的话是否会导致花费更高？
2. system role 未来几周应该是会被 OpenAI 支持的，不如届时去掉过滤？

请告知你的想法，我现在也有些纠结这个问题；另外我 push 了一个修改超时的 commit，你看看是否时间给的过长或有没有必要为了 O1 延长阈值？

Comment (Dogtiti) [reply app/client/platforms/openai.ts:186]: 1.如果是考虑到未来不久可能被支持，那么可以暂时这样过滤掉，回头开放回来即可
2.超时的问题可以保持原样,基于1的考量未来速度可能也会提升

Comment (Dogtiti) [reply app/client/platforms/openai.ts:186]: 可以~

**Review by Dogtiti — APPROVED** [review]

**Pull request merged**